pkgs/os-specific/linux/firmware/sof-firmware/default.nix

```diff
@@ -3,18 +3,19 @@
 with lib;
 stdenv.mkDerivation rec {
   pname = "sof-firmware";
-  version = "1.7";
+  version = "1.8";
 
   src = fetchFromGitHub {
     owner = "thesofproject";
     repo = "sof-bin";
     rev = "v${version}";
-    sha256 = "sha256-Z0Z4HLsIIuW8E1kFNhAECmzj1HkJVfbEw13B8V7PZLk=";
+    sha256 = "sha256-NPbzDXZoZVWriV3klemX59ACnAlb357A5V/GbmzshyA=";
   };
 
   dontFixup = true; # binaries must not be stripped or patchelfed
 
   installPhase = ''
+    cd v${version}.x
```

Comment:
Starting with v1.8 upstream has a new install script: https://github.com/thesofproject/sof-bin/blob/v1.8/install.sh
Any chance we could use this?

Reply:
The script was introduced in 1.7. As stated in the README:

> You don't have to use install.sh, you can use any recursive copy of your preference. This is all what install.sh does

Having said that, we might as well use it? This seems to work:
```
      installPhase = ''    
        mkdir -p $out/lib/firmware/intel/    
        mkdir -p $out/bin/    
        export FW_DEST=$out/lib/firmware/intel    
        export TOOLS_DEST=$out/bin    
        ./install.sh v1.8.x/v1.8    
      '';
```
The script installs some additional tools, not sure what they're for. We'll need a fixup phase that strips and patches these, but not the SOF binaries.

Reply:
Oh and `rsync` is needed as a build input.

Reply:
Actually stripping and patching the firmware binaries doesn't seem to make a difference. Maybe it's just unnecessary?

If we want the tools to be thrown in with the firmware they should be added to `systemPackages` too.

Reply:
@evenbrenden why dont you take this change over then since you seem to have thoughts about how it shoudl be done.

Reply:
IMO using the install script is a nice practice, but I think it's fine to merge this PR as is.

As for the tools, I'm really not sure if we should add them, just noticed them now. Either way that could be done in a separate PR.

Reply:
The track record of the SOF install script is also pretty bad. They manage to break the script with every other release. It was me, who switched the `installPhase` to just perform a recursive copy.

Reply:
For copying two files the script is quite big and uses rsync which we don't need.